### DISCUSSION TITLE
Naming this vue-devtools when it's so different from the original browser extension may invite confusion

### DISCUSSION THREAD

**Discussion by Offlein** [vuejs/devtools#529] [General]
What is the name of this project? Is it "Vue DevTools"? Because that's [definitely something that already existed](https://devtools.vuejs.org/). (And it was, I think, made by the same people as this?) And if you Google for "Vue DevTools" you get all sorts of results about that thing -- not this thing.

This Vue DevTools is decidedly not the same thing as the thing that used to be Vue DevTools. "Vue DevTools" the original was a browser plugin that was well-known and commonly used by most Vue developers.

_This_ thing is something that does the same stuff, but it requires you to install a plugin in your Vue project, and it doesn't exist as a plugin. (Or it does have an optional one, but it's in Beta state for Chrome right now and doesn't exist for Firefox.) So it's plainly a different "thing".

This is apparently the next iteration of what everyone used to know as "Vue DevTools" but by virtue of it functioning completely differently it's not just "the next version of Vue DevTools". And even if it were, the current website for, the, uh, "other" Vue DevTools makes no mention of THIS Vue DevTools.

I just don't understand what the benefit of making a brand new, seemingly complete rewrite of "Vue DevTools" is and then calling it "Vue DevTools" but keeping the other "Vue DevTools" around.

At least the package name is called "devtools-next". It's not 100% perfect in my mind, but it's better than the alternative. If we're dead set on keeping this "Vue DevTools" maybe at least we could change the documentation to called it "Vue DevTools Next" at least?

It seems like it would still be clearer if this were called "Vue NextTools" or "Vue UberTools" or "Vue3 Devtools" or something, but I'd be happy just to be able to have conversations about this software without having a lengthy introductory discussion each time. "Are you using Vue DevTools yet? Not the Vue DevTools that you definitely have been using, but the new one called the same thing that's completely different."

**Comment by WebMechanic** [comment]
As written on the tin: "The **next iteration** of Vue DevTools". 
Vue 3 once started as `vue-next' while the codebase was in flux and incompatible. If a developer decides to use "a next thing" they should understand that it will be something different that builds upon the former thing. Some people also call this evolution.

>  "Vue DevTools" the original was a browser plugin that was well-known and commonly used by most Vue developers.

and that still holds true esp. for Vue2 dev who cannot use _this thing_ 'cos _this thing_ is Vue3 only. Evolution, you know?
With Vue2 EOL it doesn't make sense to keep the code in _that thing_ clogging up resources, hence _this thing_ will eventually take over when it's safe for developers to do the switch to Vue3. 
Or maybe it'll eventually become the de facto "**Vue3 DevTools**" and the "original" remains the "**Vue DevTools**."

Maybe relax and let this mature before bitching about it's name? 
Have a look at the [roadmap](https://github.com/vuejs/devtools-next/discussions/112) discussion as it may answer some questions like why development of it is more slow paced.

Have a nice time.

**Reply by Offlein** [reply]
Thanks for the reply! It feels oddly aggressive but my guess is I must've came off that way to you, so my apologies for that. Maybe I will remove the word "incredibly" from the title.

Anyway I'm certainly not concerned about the speed of development. I think the people most active in this project are wonderful.

>As written on the tin: "The next iteration of {XYZ}".

That also seems like something that might reasonably be written in the changelog of any project when a new release comes out.

Anyway, if nobody else is confused by the project name that's wonderful. For me, I just had the minor hassle of explaining it to 2 other developers who seemed to feel similarly.  It was a minor hassle, but the fundamental confusion is simply, "Why invite a minor hassle when a solution is available?" Call it "Vue3 Super Devtools" -- or anything else! Call it "Colada" ...or "Margarita" or something! -- and it's a non-issue.

And if the answer is "Well, it's already named and the minor hassle isn't worth renaming it" that's certainly a reasonable take as well.

**Reply by WebMechanic** [reply]
> It feels oddly aggressive but my guess is I must've came off that way to you

yes it was a bit intended on my side 'cos your post sounded somewhat _condescending_ by repeatedly using the expressions _this thing_ and _that thing_  :-)

However, following you post I learned that the browser extension (still) handles Vue2, so I am somewhat with you. Naming this new beast simply "Vue3 DevTools" or "Vue DevTools NG" might be sufficient to communicate it's (a) different and (b) for use with Vue3 only. And yes: the home page should probably explain that there's _that_ browser extension and _this thing_ ... ;-)

Have a nice time.
